stdlib/string.gr

```diff
@@ -1867,30 +1867,58 @@ export let forEachCodePointi = (fn: (Number, Number) -> Void, str: String) => {
   void
 }
 
+@unsafe
 let trimString = (str: String, end: Bool) => {
-  let chars = explode(str), charsLength = length(str)
-  let mut i = 0, offset = 1
+  let (>>>) = WasmI32.shrU
+  let (+) = WasmI32.add
+  let (*) = WasmI32.mul
+  let (-) = WasmI32.sub
+  let (<) = WasmI32.ltU
+  let (==) = WasmI32.eq
+  let (!=) = WasmI32.ne
+
+  let mut stringPtr = WasmI32.fromGrain(str)
+  let charsLength = WasmI32.load(stringPtr, 4n)
+  stringPtr += 8n
+  let mut i = 0n, offset = 1n
   if (end) {
-    i = charsLength - 1
-    offset = -1
+    i = charsLength - 1n
+    offset = -1n
   }
-  for (; i < charsLength && i > -1; i += offset) {
-    let currentChar = chars[i]
+  let mut count = 0n
+  for (; i < charsLength; i += offset) {
+    // Get The Byte
+    let byte = WasmI32.load8U(stringPtr, i)
     // TODO: Use unicode whitespace property and unicode line terminator once github issue #661 is completed
+    // Check Multi-Byte Characters
+    if (byte == 0xEFn && WasmI32.load(stringPtr, i - 1n) >>> 8n == 0xBFBBEFn) {
+      i += offset * 2n
```

Comment:
Can you explain this?

Reply:
I can add a comment if you want explaining bassically we want to move the index forward so we are not scanning half the Character but we might be moving backwardss or forwards in the loop so we use `offset * 2` to get the sign.

Reply:
It is not `i +=2` because when we call `trimEnd`, `trimString` goes in reverse